app/controllers/facility_accounts_controller.rb

```diff
@@ -52,7 +52,10 @@ def initialize
 
   # GET /facilties/:facility_id/accounts
   def index
-    @accounts = Account.has_orders_for_facility(current_facility).paginate(:page => params[:page])
+    accounts = Account.with_orders_for_facility(current_facility)
+    accounts = accounts.where(facility_id: nil) if current_facility.cross_facility?
```

Comment:
I think this is fine. I'm not sure a better way. It's already a little funky because `ids_with_orders` adds some extra filtering if it's `single_facility?`